src/libraries/Hooks.sol

```diff
@@ -261,16 +261,16 @@ library Hooks {
             if (key.fee.isDynamicFee()) lpFeeOverride = result.parseFee();
 
             // skip this logic for the case where the hook return is 0
-            if (self.hasPermission(BEFORE_SWAP_RETURNS_DELTA_FLAG)) {
-                hookReturn = BeforeSwapDelta.wrap(result.parseReturnDelta());
+            if (self.hasPermission(BEFORE_SWAP_RETURNS_DELTAS_FLAG)) {
+                hookReturn = BeforeSwapDeltas.wrap(result.parseReturnDeltas());
 
                 // any return in unspecified is passed to the afterSwap hook for handling
-                int128 hookDeltaSpecified = hookReturn.getSpecifiedDelta();
+                int128 hookDeltaspecified = hookReturn.getSpecifiedDelta();
```

Comment:
this should just be hookDeltaSpecified bc it is only concerning one delta in the specified token